spv/sync.go

```diff
@@ -125,7 +138,86 @@ func (s *Syncer) unsynced() {
 	if atomic.CompareAndSwapUint32(&s.atomicWalletSynced, 1, 0) &&
 		s.notifications != nil &&
 		s.notifications.Synced != nil {
-		s.notifications.Synced(false)
+		s.notifications.Unsynced()
+	}
+}
+
+// peerConnected updates the notification for peer count, if set.
+func (s *Syncer) peerConnected() {
+	if s.notifications != nil && s.notifications.PeerConnected != nil {
+		s.notifications.PeerConnected(int32(len(s.remotes)))
```

Comment:
I dislike how this assumes the remotes mutex is held, especially without any any comment indicating so.  I'd rather this count be passed as parameters.  After doing this, we can reduce the locking by releasing the mutex first and then running the notification callback.